File: banking/custom/bank.js
frappe.ui.form.on("Bank", {
	refresh: function (frm) {
		if (frm.doc.ebics_host_id && frm.doc.ebics_url) {
			frm.add_custom_button(
				__("Show Protocol Versions"),
				() => {
					frappe.call({
						method: "banking.custom.bank.get_protocol_versions",
						args: {
							bank_name: frm.doc.name,
						},
						callback: (r) => {
							frappe.msgprint({
								title: __("EBICS Protocol Versions"),
								message: JSON.stringify(r.message, null, 2),
							});
						},
					});
				},
				"EBICS"
			);
		}
	},
});

File: banking/custom/bank.py
import frappe
from frappe import _

from banking.ebics.utils import get_protocol_versions as _get_protocol_versions


@frappe.whitelist()
def get_protocol_versions(bank_name: str):
	bank = frappe.get_doc("Bank", bank_name)
	bank.check_permission("write")

	if not bank.ebics_host_id or not bank.ebics_url:
		frappe.throw(
			_("Bank {0} does not have EBICS Host ID or URL set").format(bank_name),
			title=_("EBICS Configuration Missing"),
		)

	return _get_protocol_versions(bank.ebics_host_id, bank.ebics_url)
